Python Scripts & Updates

YuurinBee opened the pull request.

j00v commented:
I've tested the compilation with the updated Makefile as well, both Qt binaries compiled nicely

YuurinBee committed:
Github Workflows + Python Update

Setup GitHub Workflows, added folder and two .yml files, updated Python version.

YuurinBee committed:
[Tests] Remove redundant function

YuurinBee committed:
scripted-diff: test: Use py3.5 bytes::hex() method

-BEGIN VERIFY SCRIPT-
sed -i -e "s/def bytes_to_hex_str/def b_2_x/g" $(git grep -l bytes_to_hex_str)

export RE_B_0="[^()]*"                          # match no bracket
export RE_B_1="${RE_B_0}\(${RE_B_0}\)${RE_B_0}" # match exactly one ()
export RE_B_2="${RE_B_0}\(${RE_B_1}\)${RE_B_0}" # match wrapped (())

export RE_M="(b2x|bytes_to_hex_str)\(((${RE_B_0}|${RE_B_1}|${RE_B_2})*)\)"

sed -i --regexp-extended -e "s/${RE_M}/\2.hex()/g"      $(git grep -l -E '(b2x|bytes_to_hex_str)')

sed -i --regexp-extended -e "/  +bytes_to_hex_str( as b2x)?,/d"    $(git grep -l bytes_to_hex_str)
sed -i --regexp-extended -e "s/ +bytes_to_hex_str( as b2x)?,//g"   $(git grep -l bytes_to_hex_str)
sed -i --regexp-extended -e "s/, bytes_to_hex_str( as b2x)?//g"    $(git grep -l bytes_to_hex_str)

export RE_M="(binascii\.)?hexlify\(((${RE_B_0}|${RE_B_1}|${RE_B_2})*)\).decode\(${RE_B_0}\)"

sed -i --regexp-extended -e "s/${RE_M}/\2.hex()/g" $(git grep -l hexlify -- ':(exclude)share')

sed -i --regexp-extended -e  "/from binascii import hexlify$/d" $(git grep -l hexlify -- ':(exclude)share')
sed -i --regexp-extended -e "s/(from binascii import) .*hexlify/\1 unhexlify/g" $(git grep -l hexlify -- ':(exclude)share')

sed -i -e 's/ignore-names "/ignore-names "b_2_x,/g' ./test/lint/lint-python-dead-code.sh
-END VERIFY SCRIPT-

YuurinBee committed:
Add Lint Files

Added lint folder, files, scripts, etc.

YuurinBee committed:
scripts: clean github-merge.py and symbol-check.py

clean both files to adhere to flake8 standards.

YuurinBee committed:
scripts: additional E275 fixups

YuurinBee committed:
scripted-diff: test: Remove brackets after assert

-BEGIN VERIFY SCRIPT-
sed -i --regexp-extended -e 's/assert ?\((.+)\)(( )*)?(#.*)?$/assert \1\3\3\4/g' $(git grep -l --extended-regexp 'assert ?\(' test)
-END VERIFY SCRIPT-

YuurinBee committed:
tests: Avoid using mutable default parameter values

YuurinBee committed:
scripts: exclude wallet_bumpfee.py from linting

currently unsupported test

YuurinBee committed:
scripts: remove outdated python3 workaround in scrypt.py

YuurinBee committed:
scripts: remove NONFATAL from security-check.py

YuurinBee committed:
scripts: no-longer check for 32 bit windows in security-check.py

YuurinBee committed:
add type hints to the test framework's script.py

YuurinBee committed:
Strictly enforce instance attrs in critical functional test classes.

Additionally, removed redundant parentheses and added PEP-8 compliant
spacing around those classes.

YuurinBee committed:
Create feature_dbcrash.py

YuurinBee committed:
Update test_runner.py
